src/main/views/MattermostView.ts

```diff
@@ -96,7 +93,64 @@ export class MattermostView extends EventEmitter {
         WebRequestManager.onResponseHeaders(this.extractCookies, this.view.webContents.id);
 
         // Websocket
-        WebRequestManager.onRequestHeaders(this.addOriginForWebsocket);
+        WebRequestManager.onRequestHeaders(this.addOriginForWebsocket, this.view.webContents.id);
+
+        // CORS
+        this.corsHeaders = [];
+        this.corsMethods = [];
+        WebRequestManager.onRequestHeaders(this.extractCORSHeaders, this.view.webContents.id);
+        WebRequestManager.onResponseHeaders(this.addCORSResponseHeader, this.view.webContents.id);
+    }
+
+    private extractCORSHeaders = (details: OnBeforeSendHeadersListenerDetails) => {
+        if (!details.url.match(new RegExp(`${this.tab.server.url.origin}/(.+)`))) {
```

Comment:
The RegExp needs to be anchored to the beginning of the string, otherwise a request to a crafted URL like `https://example.com/?https://community.mattermost.com/` matches. Dots and square brackets (valid in IPv6 URLs) in the origin also need to be escaped. This applies everywhere where this type of check is being used.

Also personally I would prefer `RegExp.prototype.test` over `String.prototype.match` where a boolean is expected.

Really this shouldn't be using RegExps at all, though.